rdrf/rdrf/features/steps.py

```diff
@@ -183,6 +243,43 @@ def enter_cde_on_form(step, cde_value, form, section, cde):
 
     raise Exception("could not find cde %s" % cde)
 
+@step('I enter value "(.*)" for form "(.*)" section "(.*)" cde "(.*)" in item (\d+)')
+def enter_cde_on_form_multisection(step, cde_value, form, section, cde, item):
+    formset_number = int(item) - 1
+    formset_string = "-%s-" % formset_number
+
+    def correct_item(input_element):
+        input_id = input_element.get_attribute("id")
+        return formset_string in input_id
+    
+    
+    location_is(step, form)  # sanity check
+
+    form_block = world.browser.find_element_by_id("main-form")
+    section_div_heading = form_block.find_element_by_xpath(
+        ".//div[@class='panel-heading'][contains(., '%s')]" % section)
+    section_div = section_div_heading.find_element_by_xpath("..")
+
+    label_expression = ".//label[contains(., '%s')]" % cde
+
+    for label_element in section_div.find_elements_by_xpath(label_expression):
+        input_div = label_element.find_element_by_xpath(".//following-sibling::div")
+        try:
+            input_element = input_div.find_element_by_xpath(".//input")
+            if not correct_item(input_element):
```

Comment:
same as above, I would invert he logic here to remove the continue and return based on an if